File: src/main/java/org/blogsphere/blog/Payload/Response/UserResponse.java
package org.blogsphere.blog.Payload.Response;

import java.time.LocalDateTime;
import java.util.Set;

import org.blogsphere.blog.Enum.ERole;

import lombok.Builder;

@Builder
public record UserResponse(
    String id,

    String username,

    String email,

    String firstName,

    String lastName,

    Set<ERole> roles,

    String profilePicture,

    String bio,

    LocalDateTime createdAt,

    LocalDateTime updatedAt
) {
    
}

File: frontend/src/app/models/user-response.ts
import { ERole } from "./role";

export interface UserResponse {
    id: String;
    username: String;
    email: String;
    firstName: String;
    lastName: String;
    roles: ERole[];
    profilePicture: String;
    bio: String;
    createdAt: Date;
    updatedAt: Date;
}
